components/Server.coffee

```diff
@@ -16,24 +16,30 @@ exports.getComponent = ->
       error:
         datatype: 'object'
 
+  c.autoOrdering = false
   c.servers = []
+  c.context = null
 
-  c.on 'end', ->
+  c.tearDown = (done) ->
     for server in c.servers
       server._connections = 0
       server.close()
+    c.context.deactivate()
+    c.emit 'end'
+    c.started = false
+    done()
 
   c.forwardBrackets =
     port: ['app', 'error']
 
-  c.process (input, output) ->
-    return unless input.has 'port'
+  c.process (input, output, context) ->
+    return unless input.hasData 'port'
     port = input.getData 'port'
+    c.context = context
```

Comment:
Since we support server per scope, we should probably also keep a context per scope instead of just one?